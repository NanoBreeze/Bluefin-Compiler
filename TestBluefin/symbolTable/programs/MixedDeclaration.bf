struct First {
	int b;
	int c;
} ;

First f() {
	bool b;
	string c;
}

struct Second {
	First a;
	string b;
};

Second g(First d) {
	int e;
	float f;
	First a;
	Second b;
}
