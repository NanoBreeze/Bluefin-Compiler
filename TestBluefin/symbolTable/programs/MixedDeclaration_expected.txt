declare - First - c_struct - t_First
enterScope - Level 1
resolve - int - c_builtinType - t_int
declare - b - c_var - t_int
resolve - int - c_builtinType - t_int
declare - c - c_var - t_int
exitScope - Level 1
resolve - First - c_struct - t_First
declare - f - c_function - t_First
enterScope - Level 1
resolve - bool - c_builtinType - t_bool
declare - b - c_var - t_bool
resolve - string - c_builtinType - t_string
declare - c - c_var - t_string
exitScope - Level 1
declare - Second - c_struct - t_Second
enterScope - Level 1
resolve - First - c_struct - t_First
declare - a - c_var - t_First
resolve - string - c_builtinType - t_string
declare - b - c_var - t_string
exitScope - Level 1
resolve - Second - c_struct - t_Second
declare - g - c_function - t_Second
enterScope - Level 1
resolve - First - c_struct - t_First
declare - d - c_var - t_First
resolve - int - c_builtinType - t_int
declare - e - c_var - t_int
resolve - float - c_builtinType - t_float
declare - f - c_var - t_float
resolve - First - c_struct - t_First
declare - a - c_var - t_First
resolve - Second - c_struct - t_Second
declare - b - c_var - t_Second
exitScope - Level 1
